server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1547,146 +1720,243 @@ public void regenerateCertificatesOf(Owner owner, String productId, boolean lazy
         }
     }
 
+    @Override
+    public void revokeEntitlements(List<Entitlement> entsToRevoke) {
+        revokeEntitlements(entsToRevoke, true);
+    }
+
     /**
-     * Remove the given entitlement and clean up.
+     * Revokes the given set of entitlements.
      *
-     * @param entitlement entitlement to remove
-     * @param regenModified should we look for modified entitlements that are affected
-     * and regenerated. False if we're mass deleting all the entitlements for a consumer
-     * anyhow, true otherwise. Prevents a deadlock issue on mysql (at least).
+     * @param entsToRevoke entitlements to revoke
+     * @param regenCertsAndStatuses if this revocation should also trigger regeneration of certificates
+     * and recomputation of statuses. For performance reasons some callers might
+     * choose to set this to false.
      */
     @Transactional
-    void removeEntitlement(Entitlement entitlement, boolean regenModified) {
-        Consumer consumer = entitlement.getConsumer();
-        Pool pool = entitlement.getPool();
-
-        // Similarly to when we add an entitlement, lock the pool when we remove one, too.
-        // This won't do anything for over/under consumption, but it will prevent
-        // concurrency issues if someone else is operating on the pool.
-        pool = poolCurator.lockAndLoad(pool);
-        consumer.removeEntitlement(entitlement);
+    public void revokeEntitlements(List<Entitlement> entsToRevoke, boolean regenCertsAndStatuses) {
+        if (log.isDebugEnabled()) {
+            log.debug("Starting batch revoke of entitlements: {}", getEntIds(entsToRevoke));
+        }
+        if (entsToRevoke.isEmpty()) {
```

Comment:
Is it possible/reasonable for this collection to be null? If so, we need a null check here.